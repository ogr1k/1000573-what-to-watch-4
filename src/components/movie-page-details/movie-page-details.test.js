import React from "react";
import renderer from "react-test-renderer";
import MoviePageDetails from "./movie-page-details.jsx";

const film = {
  description: `A former Prohibition-era Jewish gangster returns to the Lower East Side of Manhattan over thirty years later, where he once again must confront the ghosts and regrets of his old life.`,
  director: `Sergio Leone`,
  genre: `Crime`,
  id: 1,
  name: `Once Upon a Time in America`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/Once_Upon_a_Time_in_America.jpg`,
  rating: 9.9,
  year: 1984,
  ratings: 276395,
  starring: [`Robert De Niro`, `James Woods`, `Elizabeth McGovern`],
  video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  runTime: 500,
  isFavourite: false
};


it(`Render Movie Page Details`, () => {

  const tree = renderer
    .create(

        <MoviePageDetails
          activeTab={``}
          film={film}
          onClick={() => {}}
        />

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
