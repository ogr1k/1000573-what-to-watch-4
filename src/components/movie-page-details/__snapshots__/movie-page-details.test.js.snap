// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movie Page Details 1`] = `
Array [
  <div
    className="movie-card__poster movie-card__poster--big"
  >
    <img
      alt="Once Upon a Time in America"
      height={327}
      src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg"
      width={218}
    />
  </div>,
  <div
    className="movie-card__desc"
  >
    <nav
      className="movie-nav movie-card__nav"
    >
      <ul
        className="movie-nav__list"
      >
        <li
          className="movie-nav__item
            "
        >
          <a
            className="movie-nav__link"
            href="#"
            onClick={[Function]}
          >
            Overview
          </a>
        </li>
        <li
          className="movie-nav__item
            "
        >
          <a
            className="movie-nav__link"
            href="#"
            onClick={[Function]}
          >
            Details
          </a>
        </li>
        <li
          className="movie-nav__item
            "
        >
          <a
            className="movie-nav__link"
            href="#"
            onClick={[Function]}
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="movie-card__text movie-card__row"
    >
      <div
        className="movie-card__text-col"
      >
        <p
          className="movie-card__details-item"
        >
          <strong
            className="movie-card__details-name"
          >
            Director
          </strong>
          <span
            className="movie-card__details-value"
          >
            Sergio Leone
          </span>
        </p>
        <p
          className="movie-card__details-item"
        >
          <strong
            className="movie-card__details-name"
          >
            Starring
          </strong>
          <span
            className="movie-card__details-value"
          >
            Robert De Niro,
James Woods,
Elizabeth McGovern
          </span>
        </p>
      </div>
      <div
        className="movie-card__text-col"
      >
        <p
          className="movie-card__details-item"
        >
          <strong
            className="movie-card__details-name"
          >
            Run Time
          </strong>
          <span
            className="movie-card__details-value"
          >
            8H 20M
          </span>
        </p>
        <p
          className="movie-card__details-item"
        >
          <strong
            className="movie-card__details-name"
          >
            Genre
          </strong>
          <span
            className="movie-card__details-value"
          >
            Crime
          </span>
        </p>
        <p
          className="movie-card__details-item"
        >
          <strong
            className="movie-card__details-name"
          >
            Released
          </strong>
          <span
            className="movie-card__details-value"
          >
            1984
          </span>
        </p>
      </div>
    </div>
  </div>,
]
`;
